tilelang/cache/kernel_cache.py

```
@@ -283,59 +292,83 @@ def _save_kernel_to_disk(self, key: str, kernel: JITKernel, func: Callable = Non
-                self.logger.debug(f"Saving kernel source code to file: {device_kernel_path}")
-            if kernel.kernel_source is not None:
-                KernelCache._safe_write_file(device_kernel_path, "w", lambda file: file.write(kernel.kernel_source))
+            if self.execution_backend != "cutedsl":
+                device_kernel_path = os.path.join(cache_path, DEVICE_KERNEL_PATH)
+                if verbose:
+                    self.logger.debug(f"Saving kernel source code to file: {device_kernel_path}")
+                if kernel.kernel_source is not None:
+                    KernelCache._safe_write_file(device_kernel_path, "w", lambda file: file.write(kernel.kernel_source))
         except Exception as e:
-            self.logger.error(f"Error saving kernel source code to disk: {e}")
+            self.logger.exception(f"Error saving kernel source code to disk: {e}")
```

Ruff (0.14.8)

Redundant exception object included in `logging.exception` call

(TRY401)